File: leetcode/js/find-if-digit-game-can-be-won.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canAliceWin = function (nums) {
  let singleDigitsSum = 0;
  let doubleDigitsSum = 0;

  for (const num of nums) {
    if (num < 10) {
      singleDigitsSum += num;
    } else {
      doubleDigitsSum += num;
    }
  }

  return singleDigitsSum !== doubleDigitsSum;
};

File: leetcode/rust/find-if-digit-game-can-be-won.rs
impl Solution {
    pub fn can_alice_win(nums: Vec<i32>) -> bool {
        let mut single_digits_sum = 0;
        let mut double_digits_sum = 0;

        for num in nums {
            match num < 10 {
                true => single_digits_sum += num,
                false => double_digits_sum += num,
            };
        }

        single_digits_sum != double_digits_sum
    }
}
